snap/snapcraft.yaml

```diff
@@ -12,256 +13,318 @@ description: |
+      - edgexproxy
+    command: bin/support-notifications -confdir $SNAP_DATA/config/support-notifications/res --registry
+    daemon: simple
+    plugs: [network, network-bind]
+    stop-timeout: 1s
   support-scheduler:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh support-scheduler --registry
-    daemon: forking
+    adapter: none
+    after:
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
+    command: bin/support-scheduler -confdir $SNAP_DATA/config/support-scheduler/res --registry
+    daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
+    stop-timeout: 1s
   support-rulesengine:
+    adapter: full
+    after:
+      - export-client
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
     command: bin/support-rulesengine-wrapper.sh
     daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - export-client
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
   export-client:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh export-client --registry
-    daemon: forking
+    adapter: none
+    after:
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
+    command: bin/export-client -confdir $SNAP_DATA/config/export-client/res --registry
+    daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
+    stop-timeout: 1s
   export-distro:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh export-distro --registry
-    daemon: forking
+    adapter: full
+    after:
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
+    command: bin/export-distro -confdir $SNAP_DATA/config/export-distro/res --registry
+    daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
+    stop-timeout: 1s
   sys-mgmt-agent:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh sys-mgmt-agent --registry
-    daemon: forking
-    plugs: [network, network-bind, daemon-notify]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
-  device-modbus:
-    command: bin/go-services-wrapper.sh device-modbus
+    adapter: none
+    after:
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
+    command: bin/sys-mgmt-agent -confdir $SNAP_DATA/config/sys-mgmt-agent/res --registry
     daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
-  device-mqtt:
-    command: bin/go-services-wrapper.sh device-mqtt
+  device-random:
+    adapter: none
+    after:
+      - core-config-seed
+      - mongo-worker
```

Comment:
Is this a no-op if redis is enabled?

Reply:
Yes it is a no-op if `mongo-worker` has been disabled. Specifically, `after` in snap.yaml and snapcraft.yaml corresponds to the `After` directive in a systemd unit which doesn't do anything if the specified service is disabled.